src/hb-ot-metrics.cc

```diff
@@ -58,53 +57,48 @@ hb_ot_metrics_get (hb_font_t       *font,
+  case HB_OT_METRICS_SUBSCRIPT_EM_Y_SIZE:         return GET_METRICS (OS2, ySubscriptYSize, y);
+  case HB_OT_METRICS_SUBSCRIPT_EM_X_OFFSET:       return GET_METRICS (OS2, ySubscriptXOffset, x);
+  case HB_OT_METRICS_SUBSCRIPT_EM_Y_OFFSET:       return GET_METRICS (OS2, ySubscriptYOffset, x);
+  case HB_OT_METRICS_SUPERSCRIPT_EM_X_SIZE:       return GET_METRICS (OS2, ySuperscriptXSize, x);
+  case HB_OT_METRICS_SUPERSCRIPT_EM_Y_SIZE:       return GET_METRICS (OS2, ySuperscriptYSize, y);
+  case HB_OT_METRICS_SUPERSCRIPT_EM_X_OFFSET:     return GET_METRICS (OS2, ySuperscriptXOffset, x);
+  case HB_OT_METRICS_SUPERSCRIPT_EM_Y_OFFSET:     return GET_METRICS (OS2, ySuperscriptYOffset, y);
+  case HB_OT_METRICS_STRIKEOUT_SIZE:              return GET_METRICS (OS2, yStrikeoutSize, y);
+  case HB_OT_METRICS_STRIKEOUT_OFFSET:            return GET_METRICS (OS2, yStrikeoutPosition, y);
+  case HB_OT_METRICS_UNDERLINE_SIZE:              return GET_METRICS (post->table, underlineThickness, y);
+  case HB_OT_METRICS_UNDERLINE_OFFSET:            return GET_METRICS (post->table, underlinePosition, y);
 #undef GET_METRICS
-  case HB_OT_METRICS_GASP_RANGE_0:                return _get_gasp (font, 0, position);
-  case HB_OT_METRICS_GASP_RANGE_1:                return _get_gasp (font, 1, position);
-  case HB_OT_METRICS_GASP_RANGE_2:                return _get_gasp (font, 2, position);
-  case HB_OT_METRICS_GASP_RANGE_3:                return _get_gasp (font, 3, position);
-  case HB_OT_METRICS_GASP_RANGE_4:                return _get_gasp (font, 4, position);
-  case HB_OT_METRICS_GASP_RANGE_5:                return _get_gasp (font, 5, position);
-  case HB_OT_METRICS_GASP_RANGE_6:                return _get_gasp (font, 6, position);
-  case HB_OT_METRICS_GASP_RANGE_7:                return _get_gasp (font, 7, position);
-  case HB_OT_METRICS_GASP_RANGE_8:                return _get_gasp (font, 8, position);
-  case HB_OT_METRICS_GASP_RANGE_9:                return _get_gasp (font, 9, position);
-#undef GET_GASP
+  // case HB_OT_METRICS_GASP_RANGE_0:                return _get_gasp (font, 0, position);
+  // case HB_OT_METRICS_GASP_RANGE_1:                return _get_gasp (font, 1, position);
+  // case HB_OT_METRICS_GASP_RANGE_2:                return _get_gasp (font, 2, position);
+  // case HB_OT_METRICS_GASP_RANGE_3:                return _get_gasp (font, 3, position);
+  // case HB_OT_METRICS_GASP_RANGE_4:                return _get_gasp (font, 4, position);
+  // case HB_OT_METRICS_GASP_RANGE_5:                return _get_gasp (font, 5, position);
+  // case HB_OT_METRICS_GASP_RANGE_6:                return _get_gasp (font, 6, position);
+  // case HB_OT_METRICS_GASP_RANGE_7:                return _get_gasp (font, 7, position);
+  // case HB_OT_METRICS_GASP_RANGE_8:                return _get_gasp (font, 8, position);
+  // case HB_OT_METRICS_GASP_RANGE_9:                return _get_gasp (font, 9, position);
```

Comment:
When commenting out a block like this, please use `#if 0`.

Reply:
Done